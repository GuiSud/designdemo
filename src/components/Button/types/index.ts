export interface ButtonProps {
  children: any;
  [propsName: string]: any;
}