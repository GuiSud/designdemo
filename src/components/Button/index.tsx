/*
 * @Author: 归宿
 * @Date: 2022-11-18 11:41:48
 * @Description: 
 */
import React from 'react';
import { ButtonProps } from './types';
// import './index.less';

const Button: React.FC<ButtonProps> = (props: ButtonProps) => {
  const { children, ...args } = props;
  return (
    <>
      <button {...args}>{children}</button>
    </>
  )
}

export default Button;