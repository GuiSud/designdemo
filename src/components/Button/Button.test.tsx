/*
 * @Author: 归宿
 * @Date: 2022-11-21 14:43:34
 * @Description:
 */
// const sum = (a: number, b: number): number => a + b;

// test('adds 1 + 2 to equal 3', () => {
//   expect(sum(1, 2)).toBe(3);
// })
import React from 'react';
import renderer from 'react-test-renderer';
import Button from './index';

it('changes the class when hovered', () => {
  const component = renderer.create(
    <Button onClick={() => { console.log(this) }}>xx</Button>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  renderer.act(() => {
    if (tree && !Array.isArray(tree)) {
      tree.props.onClick();
    }
  });

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})