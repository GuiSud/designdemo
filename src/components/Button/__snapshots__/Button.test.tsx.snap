// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<button
  onClick={[Function]}
>
  xx
</button>
`;

exports[`changes the class when hovered 2`] = `
<button
  onClick={[Function]}
>
  xx
</button>
`;
